### PR TITLE
Fixes and improvements for storage queries

### PR DESCRIPTION
**Pull request by vgeddes**
Changes:
* Add support for the `state_queryStorageAt` RPC.
* Add `CreateStorageKeyPrefix` function for creating a key prefix for keys of a storage map.
* Fix JSON unmarshalling of `StorageChangeSet` RPC response to ensure we can determine whether a storage key actually has data associated with it.
* Updated tests as necessary

### CODE REVIEW

File: types/storage_change_set.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Go Substrate RPC Client (GSRPC) provides APIs and types around Polkadot and any Substrate-based chain RPC calls
 //
 // Copyright 2019 Centrifuge GmbH
+// Copyright 2021 Snowfork
```
Comment (philipstanislaus) [inline types/storage_change_set.go:4]: Should this be in other files we modified too? @musnit

File: types/storage_change_set.go [hunk 2]
```diff
@@ -28,45 +29,64 @@ type StorageChangeSet struct {
...
+			if err != nil {
+				return err
+			}
+			r.StorageData.value = data
+		} else {
+			r.StorageData.hasValue = false
+		}
 	default:
-		return fmt.Errorf("expected 1 or 2 entries for KeyValueOption, got %v", len(tmp))
+		return fmt.Errorf("expected 2 entries for StorageChange, got %v", len(tmp))
```
Comment (philipstanislaus) [inline types/storage_change_set.go:78]:
```suggestion
		return fmt.Errorf("expected 2 entries for KeyValueOption, got %v", len(tmp))
```

File: rpc/state/query_storage_at.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// Go Substrate RPC Client (GSRPC) provides APIs and types around Polkadot and any Substrate-based chain RPC calls
+//
+// Copyright 2019 Centrifuge GmbH
```
Comment (philipstanislaus) [inline rpc/state/query_storage_at.go:3]: Change this?

File: rpc/state/query_storage_at_test.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
+// Go Substrate RPC Client (GSRPC) provides APIs and types around Polkadot and any Substrate-based chain RPC calls
+//
+// Copyright 2019 Centrifuge GmbH
```
Comment (philipstanislaus) [inline rpc/state/query_storage_at_test.go:3]: And this?

**Review by philipstanislaus — COMMENTED** [review]
LGTM, just a few minor comments

File: types/storage_change_set.go (hunk 1 above)
Comment (musnit) [reply types/storage_change_set.go:4]: sure, or just remove it. putting copyrights on every file is a bit ridiculous

**Pull request merged**